src/View/Helper/BreadcrumbsHelper.php

```diff
@@ -57,7 +58,7 @@ public function add($title, $url = null, array $options = [])
     /**
      * {@inheritDoc}
      */
-    public function prepend($title, $url = null, array $options = [])
+    public function prepend($title, $url = null, array $options = []): \Cake\View\Helper\BreadcrumbsHelper
```

Comment:
Are those $this?
Then you must remove all typehints, chainable must never be typehinted here, not even with self/static etc works.

Reply:
you can't remove. may be only in cakephp core.
just because of this

Reply:
For stan we should fix "Constant DS not found."

Reply:
I added https://github.com/cakephp/cakephp-codesniffer/pull/243 to solve the issue on higher levels. But yeah, we need to fix the core here.

You can declare the constant in the bootstrap and then add this bootstrap file to phpstan via config. See how this is done e.g. in IdeHelper plugin.

Reply:
I opened a PR for the core: https://github.com/cakephp/cakephp/pull/13150
It was kind of long overdue :)

Reply:
Core should be fixed now. No typehint for chaining methods.

Reply:
This fix does not work. I take sometime to take a look for solution and it was not type hint issue for mehtod, but wrong typehint in docblock.

So working for core would be keep retrun self both for method and docblock,
E.g.
```
     * @return $this
     */
    public function prepend($title, $url = null, array $options = []): self
```

Reply:
Does the code contain $this as return value? Then NO, it is not self, neither in doc block nor code.
$this in docblock and no return type. See core fix and referenced code sniffer rule that can help you.

Reply:
Core fix does not work. Take a look at https://travis-ci.org/FriendsOfCake/bootstrap-ui/jobs/524151404

Reply:
Did u remove all typehints as you should? Doesnt look like it.

Reply:
I did it locally. Now i pushed this. Same result in stan check.